app/models/king.rb

```diff
@@ -1,2 +1,29 @@
 class King < Piece
+
+  def x_move_distance(new_x_position) # Finds the distance moved in the x axis
+    x_move_distance = (self.x_position - new_x_position).abs # .abs stops it from being negative
+  end
+
+  def y_move_distance(new_y_position) # Finds the distance moved in the y axis
+    y_move_distance = (self.y_position - new_y_position).abs # .abs stops it from being negative
+  end
+
+  def valid_move?(new_x_position, new_y_position, color=nil)
```

Comment:
Is there a reason for color = nil?

Reply:
Looking at this again, I think that I can get rid of that.